### PR TITLE
feat(elec): implement variable frequency generator and fix A380 electrical config

### PR DESCRIPTION
**Pull request by Gurgel100**
Fixes #[issue_no]

## Summary of Changes

This PR adds and fixes the following stuff for the electrical system of the A380:
* Add variable frequency generator
* Add support for the idg disconnect lights in the overhead
* Fix APU generators so that the correct generator is used in flight
* Fix BCRUs to not consume power at a factor of 300 too high

## Additional context

Discord username (if different from GitHub):

## Testing instructions

**A32NX**: Test if everything works as before in terms of generator and idgs.
**A380X**: Test the above changelog. Unfortunately no references.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: fbw-common/src/wasm/systems/systems/src/electrical/engine_generator.rs [hunk 1]
```diff
@@ -806,7 +971,7 @@ mod tests {
             assert!(test_bed.contains_variable_with_name("ELEC_ENG_GEN_1_IDG_IS_CONNECTED"));
         }
 
-        #[test]
+        /*#[test]
         fn starts_unstable_with_engines_off() {
```
[the remark this answers is not in the document]
Comment (Gurgel100) [reply fbw-common/src/wasm/systems/systems/src/electrical/engine_generator.rs:975]: I actually forgot that I commented that ;).

[the remark this answers is not in the document]
Comment (Gurgel100) [reply fbw-common/src/wasm/systems/systems/src/electrical/engine_generator.rs]: Added them back. And also found a bug in the stable determination.

File: fbw-common/src/wasm/systems/systems/src/electrical/engine_generator.rs [hunk 2]
```diff
@@ -115,49 +207,39 @@ impl SimulationElement for EngineGenerator {
...
-            .total_consumption_of(PotentialOrigin::EngineGenerator(self.number))
-            .get::<watt>();
-        let power_factor_correction = 0.8;
-        let maximum_true_power = 90000.;
-        self.load = Ratio::new::<percent>(
-            (power_consumption * power_factor_correction / maximum_true_power) * 100.,
-        );
+        let power_consumption =
+            report.total_consumption_of(PotentialOrigin::EngineGenerator(self.number));
+        let power_factor_correction = Ratio::new::<ratio>(0.8);
```
[the remark this answers is not in the document]
Comment (Gurgel100) [reply fbw-common/src/wasm/systems/systems/src/electrical/engine_generator.rs:218]: You mean extracting it to a const? Or do you mean calculating it (pretty much impossible)?

File: fbw-common/src/wasm/systems/systems/src/electrical/engine_generator.rs [hunk 3]
```diff
@@ -166,78 +248,108 @@ impl IntegratedDriveGenerator {
...
-            self.time_above_threshold_in_milliseconds = 0;
-            return;
+            return context.ambient_temperature();
         }
 
-        if !self.activated {
-            self.time_above_threshold_in_milliseconds = 0;
-            return;
-        }
+        let mut target_idg = corrected_n2.get::<percent>() * 1.8;
```
[the remark this answers is not in the document]
Comment (Gurgel100) [reply fbw-common/src/wasm/systems/systems/src/electrical/engine_generator.rs:294]: I didn't implement this function but I can still do it.

**Comment by Gurgel100** [comment]
Thank you @crocket63 for the review. I addressed all your comments.

**Comment by Popespice** [comment]
Quality Assurance Trainee Report

Discord          : Popespice
Object of testing: #8358 
Tier of Testing  : 1
Date             : 12/1/2024

Testing Process: 
1. Log into sim
2. Load into A32NX at any airport
3. Start up as usual, monitor voltage for APU
4. Start engines, disconnect APU, monitor voltage
5. Take off
6. Press IDG buttons in-flight to make sure GENS are d/c'd
7. Observe voltage while flying before IDG disconnect

Negatives:       
N/A

Testing Results: 
Passed

Conclusions: 
Working as expected

Media:       
N/A

**Comment by alexr4339** [comment]
Quality Assurance Tester Report

Discord : Alex R
Object of testing: #8358 
Tier of Testing : 2
Date : 14.01.2024

Testing Process:

Add variable frequency generator - Generators worked fine, indicated normal loads and APU generatos run at 400 Hz
Add support for the idg disconnect lights in the overhead - Disconnected IDB, DISC light illuminated on ovhd panel
Fix APU generators so that the correct generator is used in flight - A+B used on ground, when taking off and in the air it switched to GEN A only
Fix BCRUs to not consume power at a factor of 300 too high - Didnt find any specific BCRU depiction but all RU power consumption seemed normal with around 15 A (more or less depending in situation)

Negatives:
N/A

Testing Results:
Passed

Conclusions:
Everything seemed normal which is why I label it as passed, please contact me if further testing is required or any point was tested as desired yet:)

Media:
N/A

**Pull request merged**